lib/datadog/core/remote/client.rb

```diff
@@ -4,27 +4,32 @@
 
 require_relative 'configuration'
 require_relative 'dispatcher'
-require_relative '../../appsec/processor/rule_merger'
-require_relative '../../appsec/processor/rule_loader'
 
 module Datadog
   module Core
     module Remote
       # Client communicates with the agent and sync remote configuration
       class Client
+        class SyncError < StandardError; end
+
         attr_reader :transport, :repository, :id, :dispatcher
 
-        def initialize(transport, repository: Configuration::Repository.new)
+        def initialize(transport, capabilities, repository: Configuration::Repository.new)
```

Comment:
Reminder that I don't quite like that repository argument: a new client should always start anew with an empty repository.

Reply:
In production, that is the case.

But for testing is easer to provide a repository object, so when doing assertion of the repository object is more accessible. 

I think we could get rid of the `attr_reader : repository ` as I introduced for steep, but it looks like it complains about not having the accessor, but we can simply define it in the `*.rbs` file

Once we have everything in production running we can do a lot of cleanup